src/Symfony/Component/Cache/Adapter/ArrayAdapter.php

```diff
@@ -0,0 +1,180 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Cache\Adapter;
+
+use Psr\Cache\CacheItemInterface;
+use Psr\Cache\CacheItemPoolInterface;
+use Symfony\Component\Cache\CacheItem;
+use Symfony\Component\Cache\Exception\InvalidArgumentException;
+
+/**
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+class ArrayAdapter implements CacheItemPoolInterface
```

Comment:
I don't get why you mean by stale data: to me, data-staleness in the context of PSR-6 is defined by the expiry date. Until this date is reached, the data is not stale. After, it is. ArrayAdapter is not different from any other adapter when using this definition. The only concern I see that is specific to ArrayAdapter is garbage collecting the stale items. Do I miss something in what you said?